src/v/http/tests/rest_entity_test.cc

```diff
@@ -0,0 +1,120 @@
+      : http::rest_client::rest_entity(root) {}
+    ss::sstring resource_name() const final { return "o"; }
+
+    verb create_verb() const final { return verb::put; }
+    verb update_verb() const final { return verb::patch; }
+};
+
+TEST(rest_entity_test, test_resource_url) {
+    resource a{"foo"};
+    ASSERT_EQ(a.resource_url(), "foo/thing");
+    resource b{"foo/"};
+    ASSERT_EQ(a.resource_url(), "foo/thing");
+}
+
+TEST(rest_entity_test, test_resource_list) {
+    resource r{"x/v1/prefix"};
+    auto b = r.list().host("http://x.y").build();
+    ASSERT_TRUE(b.has_value());
+    const auto& request = b.value();
+    ASSERT_TRUE(request.method() == bh::verb::get);
+    ASSERT_EQ(request.at(bh::field::host), "x.y");
+    ASSERT_EQ(request.target(), "/x/v1/prefix/thing");
+}
+
+TEST(rest_entity_test, test_resource_optional_headers_and_qparams) {
+    resource r{"x/v1/prefix"};
+    auto b
+      = r.list({{{"x-1", "v"}}}, {{{"next", "22"}}}).host("http://x.y").build();
+    ASSERT_TRUE(b.has_value());
+    const auto& request = b.value();
+    ASSERT_EQ(request.at("x-1"), "v");
+    ASSERT_EQ(request.target(), "/x/v1/prefix/thing?next=22");
+}
+
+TEST(rest_entity_test, test_resource_create) {
+    resource r{"x/v1/prefix"};
+    auto b = r.create().host("http://x.y").build();
+    ASSERT_TRUE(b.has_value());
+    const auto& request = b.value();
+    ASSERT_TRUE(request.method() == bh::verb::post);
+    ASSERT_EQ(request.at(bh::field::host), "x.y");
+    ASSERT_EQ(request.target(), "/x/v1/prefix/thing");
+}
+
+TEST(rest_entity_test, test_resource_get) {
+    resource r{"x/v1/"};
+    auto b = r.get("id").host("http://x.y").build();
+    ASSERT_TRUE(b.has_value());
+    const auto& request = b.value();
+    ASSERT_TRUE(request.method() == bh::verb::get);
+    ASSERT_EQ(request.at(bh::field::host), "x.y");
+    ASSERT_EQ(request.target(), "/x/v1/thing/id");
+}
+
+TEST(rest_entity_test, test_resource_delete) {
+    resource r{"x"};
+    auto b = r.delete_("id").host("http://x.y").build();
+    ASSERT_TRUE(b.has_value());
+    const auto& request = b.value();
+    ASSERT_TRUE(request.method() == bh::verb::delete_);
+    ASSERT_EQ(request.at(bh::field::host), "x.y");
+    ASSERT_EQ(request.target(), "/x/thing/id");
+}
+
+TEST(rest_entity_test, test_resource_update) {
+    resource r{"x"};
+    auto b = r.update("id").host("http://x.y").build();
+    ASSERT_TRUE(b.has_value());
+    const auto& request = b.value();
+    ASSERT_TRUE(request.method() == bh::verb::post);
+    ASSERT_EQ(request.at(bh::field::host), "x.y");
+    ASSERT_EQ(request.target(), "/x/thing/id");
+}
+
+TEST(rest_entity_test, test_override_update) {
+    override_verbs r{"x"};
+    auto b = r.update("id").host("http://x.y").build();
+    ASSERT_TRUE(b.has_value());
+    const auto& request = b.value();
+    ASSERT_TRUE(request.method() == bh::verb::patch);
+    ASSERT_EQ(request.at(bh::field::host), "x.y");
+    ASSERT_EQ(request.target(), "/x/o/id");
```

Comment:
I think these can be `EXPECT_EQ` everywhere in these tests.

Reply:
changed